src/modules/core/components/UserPickerWithSearch/UserPickerWithSearch.tsx

```diff
@@ -0,0 +1,243 @@
+import compose from 'recompose/compose';
+import classNames from 'classnames';
+
+import { useField } from 'formik';
+import { AnyUser } from '~data/index';
+import { Address, SimpleMessageValues } from '~types/index';
+import { getMainClasses } from '~utils/css';
+
+import { ItemDataType, WrappedComponentAdditionalProps } from '../OmniPicker';
+import { Props as WithOmnipickerInProps } from '../OmniPicker/withOmniPicker';
+import { InputLabel } from '../Fields';
+import Icon from '../Icon';
+import UserAvatar from '~core/UserAvatar';
+
+import styles from './UserPickerWithSearch.css';
+import { ItemDefault } from '~core/SingleUserPicker';
+import Dropdown from './Dropdown';
+import withAdditionalOmniPicker from '~core/OmniPicker/withAdditionalOmniPicker';
+
+type AvatarRenderFn = (
+  address: Address,
+  user?: ItemDataType<AnyUser>,
+) => ReactNode;
+
+const MSG = defineMessages({
+  selectMember: {
+    id: 'SingleUserPicker.selectMember',
+    defaultMessage: 'Select member',
+  },
+  emptyMessage: {
+    id: 'SingleUserPicker.emptyMessage',
+    defaultMessage: 'No Colony members match that search.',
+  },
+  remove: {
+    id: 'SingleUserPicker.remove',
+    defaultMessage: 'Remove',
+  },
+  closedCaret: {
+    id: 'SingleUserPicker.closedCaret',
+    defaultMessage: 'Closed user picker',
+  },
+  openedCaret: {
+    id: 'SingleUserPicker.openedCaret',
+    defaultMessage: 'Opened user picker',
+  },
+});
+
+interface Props extends WithOmnipickerInProps {
+  /** Renders an extra button to remove selection */
+  disabled?: boolean;
+
+  /** Should render the select without a label */
+  elementOnly?: boolean;
+
+  /** Help text */
+  help?: string | MessageDescriptor;
+
+  /** Help text values for intl interpolation */
+  helpValues?: SimpleMessageValues;
+
+  /** Label text */
+  label: string | MessageDescriptor;
+
+  /** Label text values for intl interpolation */
+  labelValues?: SimpleMessageValues;
+
+  /** Html `name` attribute */
+  name: string;
+
+  /** Status text */
+  placeholder?: string | MessageDescriptor;
+
+  /** Override avatar rendering */
+  renderAvatar: AvatarRenderFn;
+
+  /** Item component for omnipicker listbox */
+  renderItem?: (user: ItemDataType<AnyUser>, selected?: boolean) => ReactNode;
+
+  /** Callback for things that happend after selection  */
+  onSelected?: (user: AnyUser) => void;
+
+  value?: AnyUser;
+
+  /** Provides value for data-test prop in the input used on cypress testing */
+  dataTest?: string;
+
+  /** Provides value for data-test prop in the item list component used on cypress testing */
+  itemDataTest?: string;
+
+  /** Provides value for data-test prop in the value of the input used on cypress testing */
+  valueDataTest?: string;
+
+  sidebarRef?: HTMLElement | null;
+}
+
+interface EnhancedProps extends Props, WrappedComponentAdditionalProps {}
+
+const displayName = 'UserPickerWithSearch';
+
+const UserPickerWithSearch = ({
```

Comment:
I agree it will reduce code repetition, but like in the previous comment, you suggested me not to change existing core components: https://github.com/JoinColony/colonyDapp/pull/3393#discussion_r879709290